src/sst/core/statapi/stat3dviz.h

```diff
@@ -0,0 +1,122 @@
+// Copyright 2009-2020 NTESS. Under the terms
+// of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Copyright (c) 2009-2020, NTESS
+// All rights reserved.
+//
+// This file is part of the SST software package. For license
+// information, see the LICENSE file in the top level directory of the
+// distribution.
+
+#ifndef _H_SST_CORE_STAT_3D_VIZ
+#define _H_SST_CORE_STAT_3D_VIZ
+
+#include "sst/core/sst_types.h"
+#include "sst/core/warnmacros.h"
+#include "sst/core/simulation.h"
+
+#include <string>
+
+static const std::string cubeKey = "cube";
+static const std::string lineKey = "line";
+
+namespace SST {
+namespace Statistics {
+
+// use common base class for the topology/geometry
+struct Shape3D {
+   enum Shape {
+     Box,
+     Line
+   };
+   Shape shape;
+
+   Shape3D(Shape s) :
+     shape(s)
+   {
+   }
+};
+struct Box3D : Shape3D {
+    Box3D(double x_origin, double y_origin, double z_origin,
+          double x_extent, double y_extent, double z_extent) :
+      Shape3D(Shape::Box),
+      x_origin_(x_origin), y_origin_(y_origin), z_origin_(z_origin),
+      x_extent_(x_extent), y_extent_(y_extent), z_extent_(z_extent)
+    {
+    }
+
+    double x_origin_;
+    double y_origin_;
+    double z_origin_;
+    double x_extent_;
+    double y_extent_;
+    double z_extent_;
+};
+
+struct Line3D : Shape3D {
+    Line3D(double x_first, double y_first, double z_first,
+           double x_second, double y_second, double z_second) :
+        Shape3D(Shape::Line),
+        x_first_(x_first), y_first_(y_first), z_first_(z_first),
+        x_second_(x_second), y_second_(y_second), z_second_(z_second)
+    {
+    }
+
+    double x_first_;
+    double y_first_;
+    double z_first_;
+    double x_second_;
+    double y_second_;
+    double z_second_;
+
+};
+
+struct Stat3DViz {
+
+    Stat3DViz(Params& params){
+        std::string shape = params.find<std::string>("shape", "");
+        if (shape == cubeKey) {
+            double x_origin = params.find<double>("x_origin", "");
```

Comment:
we should go ahead and just move to a single array parameter called "origin" and a single "size" parameter:
````cpp
std::vector<double> origin;
std::vector<double> size;
if (params.contains("origin"){
  params.find_array("origin", origin);
} else {
  //error need origin parameter
}

if (params.contains("size")){
 ....
} else {
 ....
}
````

Reply:
In the Python, you would just something like
````python
comp.addParam("origin", [1.0, 0.0, 0.0])
````

Reply:
Also we need to check if the parameter is actually given. The default behavior in SST is to return an empty default parameter (which I think is a terrible idea).

Reply:
👍 f1c24c572606221fa98efe528bb11251ddbdb6c5